js_modules/dagit/packages/core/src/ui/Tabs.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import styled from 'styled-components/macro';
+
+import {ColorsWIP} from './Colors';
+import {IconWrapper} from './Icon';
+import {FontFamily} from './styles';
+
+interface TabsProps {
+  children: Array<React.ReactElement<TabProps>>;
+  selectedTabId?: string;
+  onChange?: (selectedTabId: string) => void;
+  small?: boolean;
+}
+
+export const Tabs = styled(({selectedTabId, children, small, onChange, ...rest}) => {
+  if (!(children instanceof Array) || children.some((c) => c.type !== Tab)) {
+    throw new Error('Tabs must render Tab instances');
+  }
```

Comment:
Is TS enforcement insufficient on this?

Reply:
Unfortunately it seems so... I'll poke at this some more but I haven't be en able to get a TS error on other types of component children, only on basic primitives like string children.

<img width="521" alt="image" src="https://user-images.githubusercontent.com/1037212/134712771-08b806be-b484-400a-b8ca-bf0c3c61407b.png">